### ISSUE TITLE
nanobox-server service start...but not running  

### ISSUE THREAD

**Issue by gumatt** [nanobox-io/nanobox-pkgsrc-base#48]
I have some sort of configuration issue with osx High Sierra 10.13.2 and the latest nanobox 
`nanobox version
Nanobox Version 2.3.0-180117T2039 (4b3d14f)`

I have
`nanobox config set mount-type native`
and 
`nanobox update-images`

when I attempt to run nanobox in a project with this boxfile.yml

```
run.config:
  engine: python
  engine.config: 
    runtime: python-3.5
```

the result is: 
```
It appears you are running Nanobox for the first time.
Login to your Nanobox account:
Nanobox Username: gumatt
Nanobox Password:

✓ You've successfully logged in
Root privileges are required to start the server. Your system password may be requested...




Error   : nanobox-server service start was successful but the service is not running
Context :


Error   : exit status 1
Context :
```

I am never prompted for system password (as I was before the last osx update (at least that is what I am currently thinking is causing the problem) the nanobox log shows:

```2018-01-17 16:04:52 INFO  Starting nanobox server...
2018-01-17 16:04:52 INFO  Updating nanobox-update```

Any suggestions on where to look next to determine the problem?  Or has anyone seen a similar problem?

**Comment by pablorsk** [comment]
Hi! Try run before

```bash
nanobox server
```
Maybe you need add `sudo`. On [this issue](https://github.com/nanobox-io/nanobox/issues/427) you found more information.

**Comment by barttenbrinke** [comment]
The issue is that you are probably missing the tap/tun kext.
What does ```sudo kextutil tap.kext``` and ```sudo kextutil tun.kext``` give you?
I get:
```
Untrusted kexts are not allowed
Kext with invalid signature (-67062) denied: /Library/StagedExtensions/Library/Extensions/20B5C9F2-3A49-4940-AE15-448A62D3059E.kext
Bundle (/Library/Extensions/tun.kext) failed to validate, deleting: /Library/StagedExtensions/Library/Extensions/20B5C9F2-3A49-4940-AE15-448A62D3059E.kext
Unable to stage kext (/Library/Extensions/tun.kext) to secure location.
Untrusted kexts are not allowed
Kext with invalid signature (-67062) denied: /Library/StagedExtensions/Library/Extensions/76B0917F-C7C3-4686-94EB-5628C9CE6A84.kext
Bundle (/Library/Extensions/tap.kext) failed to validate, deleting: /Library/StagedExtensions/Library/Extensions/76B0917F-C7C3-4686-94EB-5628C9CE6A84.kext
Unable to stage kext (/Library/Extensions/tap.kext) to secure location.
Untrusted kexts are not allowed
Kext with invalid signature (-67062) denied: /Library/StagedExtensions/Users/bart/programming/opensource/tuntaposx/tuntap/B4BB9104-771F-4912-A8D5-63918B52662C.kext
Bundle (/Users/bart/programming/opensource/tuntaposx/tuntap/tap.kext) failed to validate, deleting: /Library/StagedExtensions/Users/bart/programming/opensource/tuntaposx/tuntap/B4BB9104-771F-4912-A8D5-63918B52662C.kext
Unable to stage kext (/Users/bart/programming/opensource/tuntaposx/tuntap/tap.kext) to secure location.
``` 
But as far as I know there is no way to force installation of the nanobox' versions and compiling them myself from source is now longer a working workaround for the latest version of high-sierra.

**Comment by barttenbrinke** [comment]
It seems to me the nanobox installer is incorrectly assuming the taptun pkg is correctly installed. Workaround is installing them manually from here: http://tuntaposx.sourceforge.net/download.xhtml

**Comment by andreyzenkov** [comment]
@barttenbrinke thank you! it works for me!

**Comment by alex-ross** [comment]
This solved the problem I had as well. One thing that I would like to add is that 30 seconds after the installer failed and I closed it, then macOS did ask me to approve an untrusted developer.

Fells lika an race condition here in the installer or in macOS. After I approved it and re-runned the nanobox installer everything went well.

**Comment by danhunsaker** [comment]
A delay in the OS sounds like exactly the sort of issue that would cause this problem... Thanks for your additional info!

**Comment by alex-ross** [comment]
Ny coworker got the same issue now and unfortunately the same thing I did in April did not helt him.

We had to dig deeper and could after a while find a solution.

# TLDR

1. Start macOS into Recovery mode
2. Open the Terminal (it's in the meny under utilities)
3. Execute this command `sudo spctl kext-consent add KS8XL6T9FZ`
4. Reboot macOS
5. Rerun the nanobox installer

# More info

We did find this [article](https://pikeralpha.wordpress.com/2017/08/29/user-approved-kernel-extension-loading/) which did explain how to add and list kernel extensions and how to remove and delete them.

So we could basically use this command in recovery mode to add the team id that we would like to trust.

```bash
sudo spctl kext-consent add <team id>
```

But we also did need to know the team ID.

All kext policies is in a sqlite database that can be found at `/var/db/SystemPolicyConfiguration/KextPolicy`. So we can open this and list all the extensions that are loaded using this.

```bash
$ /usr/bin/sqlite3 /var/db/SystemPolicyConfiguration/KextPolicy
sqlite> SELECT team_id, bundle_id, path FROM kext_load_history_v3;
```

It will return something like this

```
...
KS8XL6T9FZ|net.sf.tuntaposx.tap|/Library/Extensions/tap.kext
KS8XL6T9FZ|net.sf.tuntaposx.tun|/Library/Extensions/tun.kext
...
```

Here we get the Team ID that is `KS8XL6T9FZ`. The team ID will probably not change. But I leave this here if anyone needs to get the correct one if it ever changes or if anyone got a similar issue.
